spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/IcebergAdaptor.java

```diff
@@ -0,0 +1,146 @@
+        StringUtils.isNotBlank(jdbcUser),
+        "Couldn't get "
+            + GravitinoSparkConfig.GRAVITINO_JDBC_USER
+            + " from iceberg catalog properties");
+    String jdbcPasswrod = gravitinoProperties.get(GravitinoSparkConfig.GRAVITINO_JDBC_PASSWORD);
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(jdbcPasswrod),
+        "Couldn't get "
+            + GravitinoSparkConfig.GRAVITINO_JDBC_PASSWORD
+            + " from iceberg catalog properties");
+    String jdbcDriver =
+        gravitinoProperties.get(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_DRIVER);
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(jdbcDriver),
+        "Couldn't get "
+            + GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_DRIVER
+            + " from iceberg catalog properties");
+    icebergProperties.put(
+        GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_TYPE,
+        catalogBackend.toLowerCase(Locale.ENGLISH));
+    icebergProperties.put(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_URI, jdbcUri);
+    icebergProperties.put(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_WAREHOUSE, jdbcWarehouse);
+    icebergProperties.put(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_USER, jdbcUser);
+    icebergProperties.put(
+        GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_PASSWORD, jdbcPasswrod);
+    icebergProperties.put(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_DRIVER, jdbcDriver);
+    icebergProperties.put(
+        GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_INITIALIZE,
+        gravitinoProperties.getOrDefault(
+            GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_INITIALIZE, "true"));
+  }
+
+  @Override
+  public PropertiesConverter getPropertiesConverter() {
+    return new IcebergPropertiesConverter();
+  }
+
+  @Override
+  public SparkBaseTable createSparkTable(
+      Identifier identifier,
+      Table gravitinoTable,
+      TableCatalog sparkCatalog,
+      PropertiesConverter propertiesConverter) {
+    return new SparkIcebergTable(identifier, gravitinoTable, sparkCatalog, propertiesConverter);
+  }
+
+  @Override
+  public TableCatalog createAndInitSparkCatalog(
+      String name, CaseInsensitiveStringMap options, Map<String, String> properties) {
+    Preconditions.checkArgument(
+        properties != null, "Iceberg Catalog properties should not be null");
+
+    String catalogBackend = properties.get(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_BACKEND);
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(catalogBackend), "Iceberg Catalog backend should not be empty.");
+
+    HashMap<String, String> all = new HashMap<>(options);
+
+    switch (catalogBackend.toLowerCase(Locale.ENGLISH)) {
+      case GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_BACKEND_HIVE:
+        initHiveProperties(catalogBackend, properties, all);
+        break;
+      case GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_BACKEND_JDBC:
```

Comment:
could we support memory for test?

Reply:
It seems that we need to re-implement an InMemoryCatalog inherited from spark TableCatalog to proxy the Iceberg InMemoryCatalog. Can we implement it in another pr like jdbc? And we can go to unblock other PRs first. What do you think of it? cc @FANNG1

Reply:
OOH, seems no need to support inMemoryCatalog, because we couldn't get sparkTable from InmeoryCatalog properly

Reply:
okay.